source/shared/core_conn.cpp

```diff
@@ -789,11 +791,44 @@ void build_connection_string_and_set_conn_attr( _Inout_ sqlsrv_conn* conn, _Inou
             access_token_used = true;
         }
 
+        // Check if Authentication is ActiveDirectoryMSI 
+        // https://docs.microsoft.com/en-ca/azure/active-directory/managed-identities-azure-resources/overview
+        bool activeDirectoryMSI = false;
+        if (authentication_option_used) {
+            zval* auth_option = NULL;
+            auth_option = zend_hash_index_find(options, SQLSRV_CONN_OPTION_AUTHENTICATION);
+
+            char* option = Z_STRVAL_P(auth_option);
+
+            if (!stricmp(option, AzureADOptions::AZURE_AUTH_AD_MSI)) {
+                activeDirectoryMSI = true;
+
+                // There are two types of managed identities: 
+                // (1) A system-assigned managed identity: UID must be NULL
+                // (2) A user-assigned managed identity: UID defined but must not be an empty string
+                // In both cases, PWD must be NULL
+
+                bool invalid = false;
+                if (pwd != NULL) {
+                    invalid = true;
+                } else {
+                    if (uid != NULL && strnlen_s(uid) == 0) {
+                        invalid = true;
+                    }
+                }
+
+                CHECK_CUSTOM_ERROR(invalid, conn, SQLSRV_ERROR_AAD_MSI_UID_PWD_NOT_NULL ) {
+                    throw core::CoreException();
+                }
```

Comment:
What happens if the user sets the keyword to something other than ActiveDirectoryMsi? Do we just pass that straight to the ODBC driver?

Reply:
I'm not sure I understand the question. In terms of keywords check, please check the shared\core_conn.cpp line 733